src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -60,6 +59,29 @@ public static async Task RoundTripAsync()
             }
         }
 
+        [Fact]
+        public static async Task WriteLargeJson()
```

Comment:
Also, presumably, the issue will occur even without `JsonElement`, correct? Can we add a WriteAsync test that just directly writes a POCO that encapsulates a large object?

Reply:
Do you mean an object that wraps `JsonElement` ora poco like a `List<string>`?

Reply:
Both, (but I meant `poco like a List<string>`)